pkg/time/asset.go

```diff
@@ -73,34 +73,85 @@ func LoadFakeImageFromEmbedFs(filename string, symbolName string, logger logr.Lo
 
 			var rela elf.Rela64
 			for rela_section_reader.Len() > 0 {
-				err := binary.Read(rela_section_reader, elfFile.ByteOrder, &rela)
-				if err != nil {
-					return nil, errors.Wrapf(err, "read rela section rela64 entry %s", path)
+				if elfFile.Machine == elf.EM_X86_64 {
+					err := binary.Read(rela_section_reader, elfFile.ByteOrder, &rela)
+					if err != nil {
+						return nil, errors.Wrapf(err, "read rela section rela64 entry %s", path)
+					}
+
+					symNo := rela.Info >> 32
+					if symNo == 0 || symNo > uint64(len(syms)) {
+						continue
+					}
+
+					// The relocation of a X86 image is like:
+					// Relocation section '.rela.text' at offset 0x288 contains 3 entries:
+					// Offset          Info           Type           Sym. Value    Sym. Name + Addend
+					// 000000000016  000900000002 R_X86_64_PC32     0000000000000000 CLOCK_IDS_MASK - 4
+					// 00000000001f  000a00000002 R_X86_64_PC32     0000000000000008 TV_NSEC_DELTA - 4
+					// 00000000002a  000b00000002 R_X86_64_PC32     0000000000000010 TV_SEC_DELTA - 4
+					//
+					// For example, we need to write the offset of `CLOCK_IDS_MASK` - 4 in 0x16 of the section
+					// If we want to put the `CLOCK_IDS_MASK` at the end of the section, it will be
+					// len(imageContent) - 4 - 0x16
+
+					sym := &syms[symNo-1]
+					imageOffset[sym.Name] = len(imageContent)
+					targetOffset := uint32(len(imageContent)) - uint32(rela.Off) + uint32(rela.Addend)
+					elfFile.ByteOrder.PutUint32(imageContent[rela.Off:rela.Off+4], targetOffset)
+
+					// TODO: support other length besides uint64 (which is 8 bytes)
+					imageContent = append(imageContent, make([]byte, 8)...)
+
+				} else if elfFile.Machine == elf.EM_AARCH64 {
+					err := binary.Read(rela_section_reader, elfFile.ByteOrder, &rela)
+					if err != nil {
+						return nil, errors.Wrapf(err, "read rela section rela64 entry %s", path)
+					}
+
+					symNo := rela.Info >> 32
+					if symNo == 0 || symNo > uint64(len(syms)) {
+						continue
+					}
+					sym := &syms[symNo-1]
+					imageOffset[sym.Name] = len(imageContent)
+
+					targetOffset := uint32(len(imageContent)) - uint32(rela.Off) + uint32(rela.Addend)
+
+					// The relocation of a aarch64 image is like:
```

Comment:
Can we put the content of the condition `elfFile.Machine == elf.EM_X86_64` into `fake_image_amd64.go` and `elfFile.Machine == elf.EM_AARCH64` into `fake_image_arm64.go`? I'm not sure if it could work.

Reply:
looks that I commented at an incorrect line, it should be L76 and L106

Reply:
Good! I split them into two files. PTAL